lib/Sema/CSSimplify.cpp

```diff
@@ -4270,6 +4270,29 @@ bool ConstraintSystem::repairFailures(
     break;
   }
 
+  case ConstraintLocator::GenericArgument: {
+    // If any of the types is a hole, consider it fixed.
+    if (lhs->isHole() || rhs->isHole())
+      return true;
+
+    // Ignoring the generic argument because we may have a generic requirement
+    // failure e.g. `String bind T.Element`, so let's drop the generic argument
+    // path element to check and potentially record the requirement failure fix.
+    path.pop_back();
```

Comment:
The bigger question here would be whether it makes sense to actually recurse into `repairFailures` after `GenericArgument` has been dropped because it could be possible to diagnose not just requirement failures but other issues as well where `GenericArgument` could be involved, right?

Reply:
Humm, I thought about that too ... but wasn't sure if that should be a general rule or something to check on a case by case basis(maybe it doesn't make sense in some situations...). But right, I think is valid just to recurse because if anything we can add the restrictions before the recursion call anyways... I'll make the changes :)

Reply:
Alternatively let's make it recurses only if path has `AnyRequirement` in there to make it certain that repair would do a right thing.

Reply:
This way we could build this up case by case basis and wouldn't need to duplicate any logic.

Reply:
Right! Just adjusted :) 
Sorry, It took some time but was having some build issues